### ISSUE TITLE
Prevent "PX" or "Px" to be converted to lower case 

### ISSUE THREAD

**Issue by xsteadybcgo** [stylelint/stylelint#4048]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

"A feature. As I don't find the appropriate option"

> Which rule, if any, is this issue related to?

 `unit-case`  `white-blacklist`

> What CSS is needed to reproduce this issue?

```css
/*input*/
.foo {
  width: 75px; 
  border: 1PX solid #333;
}

```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "unit-case": ["lower"]
  }
}
```

> Which version of stylelint are you using?

e.g. `^10.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "CLI with `stylelint`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

e.g. "Yes, it's related to LESS maps."

> What did you expect to happen?
in my project , i use the postcss-plugin -- "postcss-pxtorem". This plugin will convert my css unit value 'px' to 'rem', and when you do not want this convertion , you should use "PX" or "Px" instead of "px".

e.g. 
```css
/*output*/
.foo {
  width: 75px; 
  border: 1PX solid #333;
}
```

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. 

```css
/*output*/
.foo {
  width: 75px; 
  border: 1px solid #333;
}
```

when intergrating with stylelintint , I don't want this tool to fix it as i hope my "px" coexit with "PX" or "Px" .

I have tried the options , for example: "unit-case" , but it does not have the ignore unit option as a second option. Such as:
```
{
  "rules": {
    "unit-case": ["lower", ignoreUnit: /px/i ]
  }
}
```

The "white-blacklist" does not support regexp expresion but only string or []string.

```json
{
  "rules": {
    "white-blacklist": [/px/i, "em", etc...]
  }
}
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

You need to [disable](https://stylelint.io/user-guide/faq/#how-do-i-turn-off-disable-or-ignore-a-rule) `unit-case`.

**Comment by xsteadybcgo** [comment]
> Thanks for the report and for using the template.
> 
> You need to [disable](https://stylelint.io/user-guide/faq/#how-do-i-turn-off-disable-or-ignore-a-rule) `unit-case`.

You are right! 

In my case,  I  integrate stylelint with prettier, my  .prettierrc config is: 
```
{
  "extends": ["stylelint-config-recommended", "stylelint-config-prettier"],
  "plugins": ["stylelint-prettier"],
  "rules": {
    "prettier/prettier": [true, {"parser": "less"}],
    "unit-case": null
  }
}
```
It still doesn't work for me,  but when I remove the  stylelint-prettier plugins option

```
{
  "extends": ["stylelint-config-recommended", "stylelint-config-prettier"],
  "rules": {
    "prettier/prettier": [true, {"parser": "less"}],
    "unit-case": null
  }
}
```

 It works well.  so, I think this issue was caused by [stylelint-prettier,](https://github.com/prettier/stylelint-prettier) not the stylelint itself. 
But why the rules option ```"unit-case" : null ``` does not overwrite the stylelint-prettier plugin rules. which one has the higher priority？

**Comment by hudochenkov** [comment]
stylelint-prettier just runs Prettier. And Prettier always lowercasing units, so there is nothing you can do. Maybe you can configure `postcss-pxtorem` to ignore what you need (`selectorBlackList` option).

**Comment by xsteadybcgo** [comment]
> stylelint-prettier just runs Prettier. And Prettier always lowercasing units, so there is nothing you can do. Maybe you can configure `postcss-pxtorem` to ignore what you need (`selectorBlackList` option).

Maybe it works , but not friendly and not what i want as I just  run _styleint_  within **precommit**.
Before commiting my code to git repo, the _stylelint_ will read the ".stylelintrc" config file first, then stylelint will work and convert "PX" to "px". **The postcss does not work at all during the precommit stage.** The postcss only works in building stage when i boot up the webpack.

I use vscode and prettier-vscode. Once I press the **command+s** key, the ide  will format the css code by prettier-vscode and ".stylelintrc config file", thus the "PX" has been converted.  

I do this so I can ensure my code was "clean" before being commited to repo.

Maybe I can  add comment ```// prettier-ignore```one by one to ignore css convertion when i press "command+s" key,  but too much work to do.

**Comment by hudochenkov** [comment]
It's Prettier who converts mixed case units to lowercase. If you need to go against Prettier, then you have to use `/* prettier-ignore */`. stylelint nor stylelint-prettier has nothing to do with it.
